### PR TITLE
feat : store and retrieve config

### PR DESCRIPTION
**Pull request by Pavan-Nambi**
Added a logging library (zap) for better logging of errors.

/claim #2

### CODE REVIEW

File: internal/client/client.go [hunk 1]
```diff
@@ -1,17 +1,202 @@
...
+	var creds Credentials
+	if err := json.Unmarshal(fileContent, &creds); err != nil {
+		logger.Error("Failed to unmarshal credentials", zap.String("file", credentialsPath), zap.Error(err))
+		return nil, fmt.Errorf("failed to unmarshal credentials: %v", err)
+	}
+
+	return &creds, nil
+}
+
+func getCredentialsFromUser() error {
```
Comment (Pavan-Nambi) [inline internal/client/client.go:111]: Whether we need this or no is subjective. as this can be done by user manually.
i'll remove this if you want me to.

**Comment by Pavan-Nambi** [comment]
we can also do something like encrypt data in that `credentials` file and decrypt but again this would annoy ppl who would like to update there tokens manually. so i thought to ask u before doing it

**Comment by Pavan-Nambi** [comment]
### If there is no file and user allows us to make file
```bash
permify-cli/cmd/permify-cli on git retrival [!] via go v1.21.5 took 3s 
x go run . permission check "ok" "ok" "ok" "ok" 
2023-12-10 11:36:38.867 INFO    client/client.go:40     Client initialized
Credentials file not found. Do you want to create one? (y/n): y
Enter Permify endpoint: idk
Enter Permify token: okay
Enter path to the certificate file: okay
Enter path to the private key file: okay
Credentials saved to /home/pavan/.permify/credentials
2023-12-10 11:36:49.265 INFO    client/client.go:78     Permify client created successfully
Error: invalid entity
Usage:
  permify-cli permission check <tenant> <entity:id> <permission> <subject:id> [flags]

Flags:
      --context string   check request context
  -h, --help             help for check

Global Flags:
      --endpoint string   permify gRPC endpoint

exit status 1

```

### If there are invalid credentials:
```bash
permify-cli/cmd/permify-cli on git retrival [!] via go v1.21.5 took 10s 
x go run . permission check "ok" "ok" "ok" "ok" 
2023-12-10 11:37:31.048 INFO    client/client.go:40     Client initialized
2023-12-10 11:37:31.048 WARN    client/client.go:62     Invalid credentials     {"invalidCredentials": {"endpoint":"idk","token":"","certPath":"okay","certKeyPath":"okay"}, "error": "token is required"}
github.com/Permify/permify-cli/internal/client.New
        /home/pavan/Documents/GitHub/permify-cli/internal/client/client.go:62
github.com/Permify/permify-cli/internal/cmd/permission.NewCheckCommand.check.func1
        /home/pavan/Documents/GitHub/permify-cli/internal/cmd/permission/check.go:34
github.com/spf13/cobra.(*Command).execute
        /home/pavan/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983
github.com/spf13/cobra.(*Command).ExecuteC
        /home/pavan/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115
github.com/spf13/cobra.(*Command).Execute
        /home/pavan/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
main.main
        /home/pavan/Documents/GitHub/permify-cli/cmd/permify-cli/permify-cli.go:20
runtime.main
        /usr/local/go/src/runtime/proc.go:267
Error: invalid credentials: token is required
Usage:
  permify-cli permission check <tenant> <entity:id> <permission> <subject:id> [flags]

Flags:
      --context string   check request context
  -h, --help             help for check

Global Flags:
      --endpoint string   permify gRPC endpoint

exit status 1

```

**Comment by Pavan-Nambi** [comment]
Hey,

@tolgaOzen Can you review this? 
if there is something that need to be added/changed let me know

thanks.

**Comment by Pavan-Nambi** [comment]
Closing this as  this is irrelevant now -  most likely

**Comment by webbdays** [comment]
@tolgaOzen @Pavan-Nambi 
whats wrong with this pr? seems good.

**Pull request closed without merging**